tests/preprocessing/test_pig_tables.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+import pandas as pd
+from cobra.evaluation.pigs_tables import generate_pig_tables
+
+class TestPigTablesGeneration:
+
+    @pytest.mark.parametrize("id_col_name", [None, "col_id"]) # test None as this is the default value in generate pig tabels 
+    def test_col_id(self, id_col_name):
```

Comment:
```suggestion
    def test_col_id(self, id_col_name: str | None): 
```
This is only possible for python 3.10 and above. I am wondering should I use this or the older version by importing typing and using typing.Union[str, None] for backwards compatibility reasons ? Or how does that actually work ?

Reply:
You could use Optional instead, which has the same functionality as Union, but is more readable.
```suggestion
    def test_col_id(self, id_col_name: Optional[str]):
```